src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -203,11 +204,12 @@ static void uct_cuda_copy_event_desc_init(ucs_mpool_t *mp, void *obj, void *chun
 static void uct_cuda_copy_event_desc_cleanup(ucs_mpool_t *mp, void *obj)
 {
     uct_cuda_copy_event_desc_t *base = (uct_cuda_copy_event_desc_t *) obj;
-    ucs_status_t status;
+    int active;
 
-    status = UCT_CUDA_FUNC(cudaEventDestroy(base->event));
-    if (UCS_OK != status) {
-        ucs_error("cudaEventDestroy Failed");
+    UCT_CUDADRV_CTX_ACTIVE(active);
+
+    if (active) {
+        UCT_CUDA_FUNC(cudaEventDestroy(base->event));
```

Comment:
@Akshay-Venkatesh why is the status check here removed?

Reply:
Actually @brminich suggested this here https://github.com/openucx/ucx/pull/4647#discussion_r364064215 and I agree that having a status check when it's already part of `UCT_CUDA_FUNC` doesn't make sense.